OMCompiler/SimulationRuntime/c/simulation/solver/kinsolSolver.c

```diff
@@ -184,6 +184,7 @@ static void resetKinsolMemory(NLS_KINSOL_DATA *kinsolData,
           "##KINSOL## In function SUNLinSol_Dense: Input incompatible.");
     }
   } else if (kinsolData->linearSolverMethod == NLS_LS_LAPACK) {
+    // phannebohm: kinsolData->J == NULL causes a segmentation fault?
```

Comment:
That's strange... 
`kinsolData->J` is allocated directly above or set to `NULL`, so I can't see a reason why it should not work.

I would suggest to debugg a bit to check what's goin on here. If that really seg faults there is something wrong.

Reply:
Okay, it is not
```C
    if (kinsolData->J == NULL) {
      printf("AHeu1: J==NULL\n");
    } else {
      printf("AHeu2: J!=NULL\n");
    }
```
that has a segmentation fault but the next line:
```C
kinsolData->linSol = SUNLinSol_LapackDense(kinsolData->y, kinsolData->J);
```

I guess we simply don't have a test case for this, since we usually use KINSOL with KLU.

Reply:
Fixed in #7859.

Reply:
Apparently not. The segmentation faults remain, so there must be some other problem. I'm not sure #7859 is the right fix.